wallet/src/testing.rs

```diff
@@ -232,6 +242,27 @@ pub async fn mint_token<'a>(
     }
     Ok(my_asset)
 }
+
+/// Return records the freezer has access to freeze or unfreeze but does not own.
+/// Will only return records with freeze_flag the same as the frozen arg.
+pub async fn find_freezable_records<'a>(
+    freezer: &CapeWallet<'a, CapeBackend<'a, ()>>,
+    frozen: FreezeFlag,
+) -> Vec<RecordInfo> {
+    let pks: HashSet<UserPubKey> = freezer.pub_keys().await.into_iter().collect();
+    let records = freezer.records().await;
+    records
+        .filter(|r| {
+            let ro = &r.ro;
+            // We own the record
+            if pks.contains(&ro.pub_key) {
```

Comment:
This should be `pks.contains(ro.asset_def.policy_ref().freezer_pub_key())`

Reply:
I may be a bit confused here but the I thought that this function would return records stored in this wallet, which we don't own.  Here I am filtering out records with one of our pub keys because we don't want freeze our own assets (I suppose we could). 

Maybe I should also filter for `ro.asset_def.policy_ref().freezer_pub_key()` in `freezer.freezer_pub_keys()`. But from looking at this code: https://github.com/EspressoSystems/seahorse/blob/main/src/lib.rs#L1443-L1468 I thought the only records the wallet would store which do not contain one of the wallets pub keys are those that are freezable by the wallet.

Reply:
I added the additional filtering for the freezer pub keys in the wallet matching the record

Reply:
Oh, sorry, I swapped the meaning `false` and `true` in my head when I was reading this. It makes sense now. I do think the extra explicit check that we can freeze is good, because the fact that the wallet only stores other user's records if they are freezable is a limitation that we plan to remove, so it will eventually store all records that are viewable, even if not freezable.